Task2.BookISBN/Services/CatalogService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Task2.Entities.BookISBN;
+
+namespace Task2.BookISBN.Models
+{
+    internal class CatalogService
+    {
+        private List<Book> _books;
+        private Book _book;
+        //an indexer for GET and POST(ADD-Assign) Book
+        public Book this[string isbn]
+        {
+            get
+            {
+                var books = _books.Where(book => book.ISBN == isbn).Select(book => book).ToList();
+                _book = books.FirstOrDefault();
+                if (_book != null)
+                {
+                    return _book;
+                }
+                throw new KeyNotFoundException();
+            }
+            set
+            {
+                if (value != null)
```

Comment:
Check isbn on setter for formatting. Also, it should match with value.ISBN. 

Ensure that catalog keeps books with valid ISBN and rejects others. 
```
var catalog = new CatalogService();
string isbnValid = "....";
var bookValid = new Book(isbnValid , ...);
catalog[isbnValid] = bookValid;

var bookReceived = catalog[isbnValid];
Console.WriteLine(bookValid == bookReceived);

string isbnInvalid = "xxxx";
var bookInvalid = new Book(isbnInvalid , ...);
try {
catalog[isbnInvalid ] = bookInvalid ;
}
catch(Exception e){
  //Check that exception thrown
}

```